nft_ingester/src/bin/slot_persister/main.rs

```diff
@@ -497,3 +560,232 @@ async fn process_slots(
+        let slot_backfill_offset =
+            higher_slot_boundary.saturating_sub(lower_slot_boundary).min(slot_backfill_offset);
+        let mut last_valid_key = lower_slot_boundary;
+        info!(%lower_slot_boundary, "Acquiring missed slots starting at last checkpoint");
+
+        // Store missing slots
+        let missing_slots = Arc::new(Mutex::new(Vec::new()));
+
+        info!(
+            "Starting to collect slots from {} to {}",
+            lower_slot_boundary,
+            lower_slot_boundary + slot_backfill_offset
+        );
+
+        // Collect slots from last persisted slot down to 0
+        in_mem_dumper.clear().await;
+
+        // Start slot collection
+        let _ = slots_collector
+            .collect_slots(
+                &BUBBLEGUM_PROGRAM_ID,
+                lower_slot_boundary + slot_backfill_offset,
+                lower_slot_boundary,
+                cancellation_token.child_token(),
+            )
+            .await;
+
+        // Get the collected slots
+        let mut collected_slots = in_mem_dumper.get_sorted_keys().await;
+        in_mem_dumper.clear().await;
+        collected_slots.retain(|s| *s > lower_slot_boundary);
+
+        if cancellation_token.is_cancelled() {
+            info!("Shutdown signal received, stopping...");
+            return None;
+        }
+
+        let total_slots_to_check = collected_slots.len() as u64;
+
+        info!(
+            "Collected {} slots in range {} to {}",
+            total_slots_to_check,
+            lower_slot_boundary,
+            lower_slot_boundary + slot_backfill_offset
+        );
+
+        if collected_slots.is_empty() {
+            return Some((
+                lower_slot_boundary,
+                lower_slot_boundary + slot_backfill_offset,
+                Vec::new(),
+            ));
+        }
+
+        // Prepare iterators for collected slots and RocksDB keys
+        let mut slots_iter = collected_slots.into_iter();
+        let mut next_slot = slots_iter.next();
+
+        let cf_handle = target_db.db.cf_handle(RawBlock::NAME).unwrap();
+        let mut db_iter = target_db.db.raw_iterator_cf(&cf_handle);
+        db_iter.seek(lower_slot_boundary.to_be_bytes());
+
+        let mut current_db_slot =
+            db_iter.key().and_then(|key_bytes| RawBlock::decode_key(key_bytes.to_vec()).ok());
+        // Verification loop
+        while let Some(slot) = next_slot {
+            if cancellation_token.is_cancelled() {
+                info!("Shutdown signal received, stopping...");
+                return None;
+            }
+
+            if let Some(db_slot) = current_db_slot {
+                match slot.cmp(&db_slot) {
+                    Ordering::Equal => {
+                        // Slot exists in RocksDB
+                        // Advance both iterators
+                        {
+                            let missing_slots_lock = missing_slots.lock().unwrap();
```

Comment:
why do we need a mutex on missing_slots? this looks to be a perfectly synchronous method in terms of access to it.

Reply:
shameless copy of the code from slot-checker is the reason. i'll add a task in our tracker to remove mutexes from both places